diff --git a/shell/cmake/DetectArchitecture.cmake b/shell/cmake/DetectArchitecture.cmake
index 9a4b855..ae07f63 100644
--- a/shell/cmake/DetectArchitecture.cmake
+++ b/shell/cmake/DetectArchitecture.cmake
@@ -21,6 +21,8 @@ function(detect_architecture symbol arch)
     endif()
 endfunction()
 
+detect_architecture("__EMSCRIPTEN__" wasm32)
+
 detect_architecture("__ARM64__" arm64)
 detect_architecture("__aarch64__" arm64)
 detect_architecture("_M_ARM64" arm64)
