diff --git a/core/hw/sh4/dyna/shil_canonical.h b/core/hw/sh4/dyna/shil_canonical.h
index a1be200..e606587 100644
--- a/core/hw/sh4/dyna/shil_canonical.h
+++ b/core/hw/sh4/dyna/shil_canonical.h
@@ -731,7 +731,7 @@ u32,f1,(f32 f1),
 	}
 	return res;
 )
-#elif HOST_CPU == CPU_ARM || HOST_CPU == CPU_ARM64
+#elif HOST_CPU == CPU_ARM || HOST_CPU == CPU_ARM64 || HOST_CPU == CPU_WASM
 shil_canonical
 (
 u32,f1,(f32 f1),
@@ -741,7 +741,7 @@ u32,f1,(f32 f1),
 	}
 	else {
 		res = (s32)f1;
-		// conversion of NaN returns 0 on ARM
+		// conversion of NaN returns 0 on ARM (and on wasm i32.trunc_sat_f32_s)
 		if (std::isnan(f1))
 			res = 0x80000000;
 	}
